datadog/dogstatsd/metrics.py

```diff
@@ -34,20 +42,36 @@ def __init__(self, name, value, tags, rate, timestamp=0):
     def aggregate(self, v):
         self.value = v
 
-    # TODO: This may be implemented once flushing aggregated metrics is supported
     def unsafe_flush(self):
-        pass
+        return {
+            'metric_type': MetricType.GAUGE,
+            'name': self.name,
+            'tags': self.tags,
+            'rate': self.rate,
+            'value': self.value,
+            'timestamp': self.timestamp
+        }
 
 
 class SetMetric(MetricAggregator):
     def __init__(self, name, value, tags, rate, timestamp=0):
         super(SetMetric, self).__init__(name, tags, rate, timestamp)
         self.data = set()
         self.data.add(value)
+        self.lock = Lock()
 
     def aggregate(self, v):
-        self.data.add(v)
+        with self.lock:
```

Comment:
Do we need this lock? as far as I can tell, this is only ever called while the associated metric lock is held